### ISSUE TITLE
Allow augmentations without DataKey.Input

### ISSUE THREAD

**Issue by maaft** [kornia/kornia#1498]
## 🚀 Feature

Sometimes you want to apply augmentations to e.g. masks the need to transform also an image.

## Alternatives

None

## Additional context

When using AugmentationSequential like this:
```python
self.random_aug = AugmentationSequential(
   K.RandomVerticalFlip(),
  data_keys=["mask"]
)
```
 following error is thrown:
`NotImplementedError: The first input must be DataKey.INPUT.`

**Comment by edgarriba** [comment]
maybe we could assume the order in case it's not specified. Besides, in that particular case if data keys you specify as input it will work because for geometric transformations images and masks are treated the same. /cc @shijianjian

**Comment by maaft** [comment]
Are they really currently treated the same for geometric augmentations? IMHO the interpolation method must differ.

One should never use other interpolation methods than `nearest` for mask, or you'll end up with non-binary values.

**Comment by edgarriba** [comment]
Agree, but for now they are treated the same, must be the user who specifies the interpolation method. I don't think we support out of the box such fine-grained automation. In principle, for such cases we recommend to create custom classes and define different behaviors per data type.

**Comment by shijianjian** [comment]
> One should never use other interpolation methods other than `nearest` for mask, or you'll end up with non-binary values.

I am not against making different default flags for different data keys, if we can have a clean implementation and a decent API.

**Comment by shijianjian** [comment]
I am thinking to introduce enums like MASK.INTEGER and MASK.FLOAT for this. So that the interpolation method will be set to nearest when the input is MASK.INTEGER. MASK.FLOAT will maintain current behaviour.

/cc @twsl

**Comment by miquelmarti** [comment]
Disregarding the fact that masks and inputs should not be treated the same, how could we allow augmentations without DataKey.Input? It seems the only reason to force having always the input datakey is to be able to get its shape when computing the parameters. Is there something else I'm missing? In such case, could we allow not having the input but ask for the shape instead? Could also get the shape from the mask but then it would not work when only passing bboxes or keypoints.

**Comment by shijianjian** [comment]
> Disregarding for now the fact that masks and inputs should not be treated the same, how could we allow augmentations without DataKey.Input? It seems the only reason to force having always the input datakey is to be able to get its shape when computing the parameters. Is there something else I'm missing? In such case, could we allow not having the input but ask for the shape instead? Could also get the shape from the mask but then it would not work when only passing bboxes or keypoints.

I think having an input shape is possibly a good solution to this. The general idea is to have a suite of parameters for the whole AugmentationSequential. As long as we can get the parameters in the first place, we can make it happen across different datakeys.

**Comment by edgarriba** [comment]
If anyone interested in this please, re-open. Thanks !
